internal/pkg/deploy/cloudformation/stack/transformers.go

```diff
@@ -141,16 +141,21 @@ func convertContainerHealthCheck(hc manifest.ContainerHealthCheck) *template.Con
 	}
 }
 
-func convertHostedZone(m manifest.RoutingRuleConfiguration) (template.AliasesForHostedZone, error) {
+func convertHostedZone(alias manifest.Alias, defaultHostedZone *string) (template.AliasesForHostedZone, error) {
 	aliasesFor := make(map[string][]string)
-	defaultHostedZone := m.HostedZone
-	if len(m.Alias.AdvancedAliases) != 0 {
-		for _, alias := range m.Alias.AdvancedAliases {
+	if len(alias.AdvancedAliases) != 0 {
+		for _, alias := range alias.AdvancedAliases {
 			if alias.HostedZone != nil {
+				if isDuplicateAliasEntry(aliasesFor[*alias.HostedZone], aws.StringValue(alias.Alias)) {
```

Comment:
let's consider an example manifest. 

```
name: lbservice
type: Load Balanced Web Service

image:
  port: 80

http:
  path: '/'
  target_port: 82
  alias:
    - name: example.com
      hosted_zone: mockHostedZone2
    - name: foobar.com
    - name: "*.foobar.com"
  hosted_zone: mockHostedZone1
  additional_rules:
    - path: "/admin"
       target_port: 81   
       alias:
         - name: adminExample.com
            hosted_zone: mockHostedZone3
          - name: foobar.com
       hosted_zone: mockHostedZone1
```
As shown in the preceding example, the alias "foobar.com" will be added to "mockHostedZone1" in both the primary and additional rules. Because customers can specify the same alias but different paths to reach different container ports, So in order to make sure that `mockHostedZone1` doesn't have a duplicate entry of `foobar.com` we need to make sure it is not already in the []string of `mockHostedZone1`

Reply:
That makes sense but i am not convinced it is the most efficient way of doing it especially we need to loop through to see if the alias is already in the list. Maybe consider using a map?